plugin/src/main/java/eu/maveniverse/maven/njord/plugin3/CheckArtifactsAvailabilityMojo.java

```diff
@@ -0,0 +1,255 @@
+            if (split.length != 2) {
+                throw new IllegalArgumentException(
+                        "Invalid alt deployment repository syntax (supported is id::url): " + remoteRepository);
+            }
+            String id = split[0];
+            String url = split[1];
+            if (mode == RepositoryMode.SNAPSHOT) {
+                result = Optional.of(new RemoteRepository.Builder(id, "default", url)
+                        .setReleasePolicy(new RepositoryPolicy(false, null, null))
+                        .build());
+            } else {
+                result = Optional.of(new RemoteRepository.Builder(id, "default", url)
+                        .setSnapshotPolicy(new RepositoryPolicy(false, null, null))
+                        .build());
+            }
+        }
+        if (result.isPresent()) {
+            result = Optional.of(repositorySystem.newDeploymentRepository(
+                    mavenSession.getRepositorySession(), result.orElseThrow(J8Utils.OET)));
+        }
+        return result;
+    }
+
+    /**
+     * Uses Resolver to perform "existence checks" for given artifacts in given remote repository target. Based on
+     * parameters it may do "one pass" or "wait/poll".
+     */
+    protected void checkAvailability(Collection<Artifact> artifacts, RemoteRepository target)
+            throws IOException, MojoFailureException {
+        Duration waitTimeout = Duration.parse(this.waitTimeout);
+        Duration waitSleep = Duration.parse(this.waitSleep);
+        Map<Artifact, Boolean> artifactsMap = artifacts.stream().collect(Collectors.toMap(a -> a, a -> false));
+
+        Instant waitingUntil = Instant.now().plus(waitTimeout);
+        AtomicInteger toCheck = new AtomicInteger(artifactsMap.size());
+        if (wait) {
+            logger.info(
+                    "Waiting for {} artifacts to become available from {} (poll {}; timeout {})",
+                    artifactsMap.size(),
+                    target.getUrl(),
+                    waitSleep,
+                    waitTimeout);
+        } else {
+            logger.info("Checking for {} artifacts availability from {}", artifactsMap.size(), target.getUrl());
+        }
+        try (RepositoryConnector repositoryConnector =
+                repositoryConnectorProvider.newRepositoryConnector(mavenSession.getRepositorySession(), target)) {
+            while (toCheck.get() > 0) {
+                logger.info("Checking availability of {} artifacts (out of {}).", toCheck.get(), artifactsMap.size());
+                List<ArtifactDownload> artifactDownloads = new ArrayList<>();
+                artifactsMap.forEach((key, value) -> {
+                    if (!value) {
+                        ArtifactDownload artifactDownload = new ArtifactDownload(key, "njord", null, null);
+                        artifactDownload.setRepositories(Collections.singletonList(target));
+                        artifactDownload.setExistenceCheck(true);
```

Comment:
I was thinking that maybe we could only test if the POM file corresponding to the artifact is available.
I think that would be enough and maybe a bit less disruptive for the Central server.

But maybe `setExistenceCheck(true)` only sends a HEAD request?

Reply:
Yes, HEAD request.

Reply:
There is no content transport happening at all.